remote-trait-object/src/context.rs

```diff
@@ -82,13 +82,20 @@ impl Context {
         Arc::downgrade(&self.port.clone().expect("It becomes None only when the context is dropped.")) as Weak<dyn Port>
     }
 
+    pub fn clear_service_registry(&mut self) {
+        self.port.as_mut().unwrap().clear_registry();
+    }
+
     pub fn disable_garbage_collection(&self) {
         self.port.as_ref().expect("It becomes None only when the context is dropped.").set_no_drop();
     }
 }
 
 impl Drop for Context {
     fn drop(&mut self) {
+        self.port.as_ref().unwrap().set_no_drop();
+        self.port.as_ref().unwrap().clear_registry();
```

Comment:
@majecty To answer your question, the new test `drop_service_which_holds_remote` will fail without these two operations. The reason is because all remote services must be dropped before the weak pointer of port is still alive. The weak pointer can't be upgraded after line 103, but without this two operations, some remote service might be destructed too late, during forwarder's destruction, which happens in line 103 too.

Reply:
OK, I understand the situation.
Some use case of remote-trait-object is that a service object has other remote service objects as fields. This PR makes service objects dropped before a port to ensure the service object drops the remote service object.

I came up with a concern that this fix will not fix all the situations that a remote service object is not dropped before Port. For example, a module's context object can have several remote service objects. They will not be dropped in the current situation. I couldn't think of a good solution to the issue.

Could you add comments that we are dropping service objects for the case when the service objects may have remote service objects? It is not an obvious situation and it is hard to find out from reading code.